patches/gtsam.patch

```diff
@@ -1,13 +1,121 @@
 diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 8c4229e..9534be1 100644
 --- ./CMakeLists.txt
 +++ ./CMakeLists.txt
-@@ -207,7 +207,7 @@ endif()
+@@ -53,7 +53,7 @@ endif()
+ 
+ # Configurable Options
+ if(GTSAM_UNSTABLE_AVAILABLE)
+-    option(GTSAM_BUILD_UNSTABLE              "Enable/Disable libgtsam_unstable"          ON)
++    option(GTSAM_BUILD_UNSTABLE              "Enable/Disable libgtsam_unstable"           OFF)
+ endif()
+ option(GTSAM_BUILD_STATIC_LIBRARY        "Build a static gtsam library, instead of shared" OFF)
+ option(GTSAM_USE_QUATERNIONS             "Enable/Disable using an internal Quaternion representation for rotations instead of rotation matrices. If enable, Rot3::EXPMAP is enforced by default." OFF)
+@@ -219,12 +219,13 @@ endif()
  ### use our patched version of Eigen
  ### See:  http://eigen.tuxfamily.org/bz/show_bug.cgi?id=704 (Householder QR MKL selection)
  ###       http://eigen.tuxfamily.org/bz/show_bug.cgi?id=705 (Fix MKL LLT return code)
 -option(GTSAM_USE_SYSTEM_EIGEN "Find and use system-installed Eigen. If 'off', use the one bundled with GTSAM" OFF)
-+option(GTSAM_USE_SYSTEM_EIGEN "Find and use system-installed Eigen. If 'off', use the one bundled with GTSAM" ON)
-
++option(GTSAM_USE_SYSTEM_EIGEN "Find and use system-installed Eigen. If 'off',use the one bundled with GTSAM" ON)
+ 
  # Switch for using system Eigen or GTSAM-bundled Eigen
  if(GTSAM_USE_SYSTEM_EIGEN)
-
+ 	find_package(Eigen3 REQUIRED)
+ 	include_directories(AFTER "${EIGEN3_INCLUDE_DIR}")
++        find_package(LAPACK REQUIRED)
+ 
+ 	# Use generic Eigen include paths e.g. <Eigen/Core>
+     set(GTSAM_EIGEN_INCLUDE_PREFIX "${EIGEN3_INCLUDE_DIR}")
+@@ -335,6 +336,10 @@ if(GTSAM_ENABLE_CONSISTENCY_CHECKS)
+   add_definitions(-DGTSAM_EXTRA_CONSISTENCY_CHECKS)
+ endif()
+ 
++if(NOT WIN32)
++    option(GTSAM_BUILD_PKGCONFIG "Build pkg-config .pc file for GTSAM" ON)
```

Comment:
Do you plan to use gtslam headers directly in orogen ? If not, there is really no reason to add a pkg-config file. (that would remove the need to patch at all ...)